lib/measured/measurable.rb

```diff
@@ -6,7 +6,7 @@ class Measured::Measurable < Numeric
   def initialize(value, unit)
     raise Measured::UnitError, "Unit value cannot be blank" if value.blank?
 
-    @unit = self.class.unit_system.to_unit_name!(unit)
+    @unit = unit.is_a?(Measured::Unit) ? unit : self.class.unit_system.unit_for!(unit)
```

Comment:
I didn't do this at first because I was worried about too large of a performance penalty, but it's small enough that I think this is a good thing 👍 

Would you like me to also revert the tests to using symbols / strings?